File: backend/src/main/java/org/certificaciones/funproeibbackend/service/UsuarioService.java
package org.certificaciones.funproeibbackend.service;

import org.certificaciones.funproeibbackend.dto.CambiarPasswordRequest;
import org.certificaciones.funproeibbackend.dto.LoginRequest;
import org.certificaciones.funproeibbackend.dto.UsuarioRegistroRequest;
import org.certificaciones.funproeibbackend.dto.UsuarioResponse;

import java.util.List;

public interface UsuarioService {
    UsuarioResponse registrar(UsuarioRegistroRequest request);
    UsuarioResponse login(LoginRequest request);
    UsuarioResponse obtenerPorId(Long id);
    List<UsuarioResponse> listarTodos();
    void verificarCorreo(String token);
    void reenviarVerificacion(String correo);
    void cambiarPassword(CambiarPasswordRequest request);
}

File: frontend/src/app/core/services/usuario.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { CambiarPasswordRequest, LoginRequest, ReenviarVerificacionRequest, UsuarioRegistroRequest, UsuarioResponse } from '../models/usuario.model';

@Injectable({ providedIn: 'root' })
export class UsuarioService {
  constructor(private api: ApiService) {}

  registrar(request: UsuarioRegistroRequest): Observable<UsuarioResponse> {
    return this.api.post<UsuarioResponse>('/usuarios/registro', request);
  }

  login(request: LoginRequest): Observable<UsuarioResponse> {
    return this.api.post<UsuarioResponse>('/usuarios/login', request);
  }

  obtenerPorId(id: number): Observable<UsuarioResponse> {
    return this.api.get<UsuarioResponse>(`/usuarios/${id}`);
  }

  listarTodos(): Observable<UsuarioResponse[]> {
    return this.api.get<UsuarioResponse[]>('/usuarios');
  }

  verificarCorreo(token: string): Observable<void> {
    return this.api.get<void>(`/usuarios/verificar-correo?token=${encodeURIComponent(token)}`);
  }

  reenviarVerificacion(request: ReenviarVerificacionRequest): Observable<void> {
    return this.api.post<void>('/usuarios/reenviar-verificacion', request);
  }

  cambiarPassword(request: CambiarPasswordRequest): Observable<void> {
    return this.api.post<void>('/usuarios/cambiar-password', request);
  }
}
